omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/field/PersonAddressWithHierarchyFragmentController.java

```diff
@@ -28,8 +28,8 @@ public void controller(FragmentModel model,
         for (AddressHierarchyLevel level : levels) {
             if (level.getAddressField() != null) {
                 validLevels.add(level);
-                List<AddressHierarchyEntry> entries = addressHierarchyService.getAddressHierarchyEntriesByLevel(level);
-                if ( entries == null || (entries !=null && entries.size() < 1)) {
+                Integer count = addressHierarchyService.getAddressHierarchyEntryCountByLevel(level);
+                if ( count == null || (count !=null || count.intValue() < 1)) {
```

Comment:
fyi @cioan ... With 2 "ors", wouldn't this always return true?  (I think the "count != null" can just be totally removed)...